pandas/core/frame.py

```diff
@@ -8531,7 +8547,12 @@ def _get_data(axis_matters):
                 raise NotImplementedError(msg)
             return data
 
-        if numeric_only is not None and axis in [0, 1]:
+        # special case for block-wise
+        if (
+            not self._mgr.any_extension_types
```

Comment:
why are we excluding any_extension_types here?  `blk_func` looks like it should handle EAs appropriately

Reply:
It indeed handles EA correctly (that fix was split off from this PR a while ago), but the reason I put this check here: the motivation for this branch (do block-wise when possible because it is faster) is not the case for EAs, as block-wise is actually slower. And secondly, that also ensures there is only a single code path when having EAs. 

Now, we could also only check for the nullable extension type (nullable int, float, bool etc, so it only checks for the new EAs), and not for all extension types in general (which then also includes things like datetimetz)

Reply:
@jbrockmendel I reverted this to keep the original check (so the same cases will take the block-wise code path as before this PR)

Reply:
thanks, ill take a fresh look